src/SFA.DAS.Roatp.CourseManagement.Web/Controllers/AddProviderContact/ProviderContactSelectStandardsController.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System.Linq;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using SFA.DAS.Roatp.CourseManagement.Web.Infrastructure;
+using SFA.DAS.Roatp.CourseManagement.Web.Infrastructure.Authorization;
+using SFA.DAS.Roatp.CourseManagement.Web.Models.ProviderContact;
+using SFA.DAS.Roatp.CourseManagement.Web.Services;
+
+namespace SFA.DAS.Roatp.CourseManagement.Web.Controllers.AddProviderContact;
+
+[Authorize(Policy = nameof(PolicyNames.HasProviderAccount))]
+[Route("{ukprn}/contact-select-standards", Name = RouteNames.ProviderContactSelectStandards)]
+public class ProviderContactSelectStandardsController(ISessionService _sessionService) : ControllerBase
+{
+    public const string ViewPath = "~/Views/AddProviderContact/SelectStandards.cshtml";
+
+    [HttpGet]
+    public IActionResult SelectStandards(int ukprn)
+    {
+        var providerContactModel = _sessionService.Get<ProviderContactSessionModel>();
+        if (providerContactModel == null) return RedirectToRoute(RouteNames.ReviewYourDetails, new { ukprn = Ukprn });
+
+        var model = new AddProviderContactStandardsViewModel
+        {
+            Standards = providerContactModel.Standards
+        };
+
+        return View(ViewPath, model);
+    }
+
+    [HttpPost]
+    public IActionResult PostStandards(int ukprn, AddProviderContactStandardsSubmitViewModel submitModel)
+    {
+
+        var providerContactModel = _sessionService.Get<ProviderContactSessionModel>();
+        if (providerContactModel == null) return RedirectToRoute(RouteNames.ReviewYourDetails, new { ukprn = Ukprn });
+
+        if (!ModelState.IsValid)
+        {
+            var viewModel = new AddProviderContactStandardsViewModel
+            {
+                Standards = providerContactModel.Standards
+            };
+
+            return View(ViewPath, viewModel);
+        }
+
+        var model = new AddProviderContactStandardsViewModel
+        {
+            Standards = providerContactModel.Standards,
+        };
+
+
+        foreach (var standard in from standard in model.Standards from item in submitModel.ProviderCourseIds where standard.ProviderCourseId == item select standard)
```

Comment:
This is not reading very well, I had to read it closely for few times to understand what is being done. Few things can be done to improve readability: 
- Rename submitModel.ProviderCourseIds to submitModel.SelectedCourseIds 
- Rename item to selectedCourseId
- Also spotted an issue, nothing is being unselected, so if I am going back and forth in the journey and changing my selections, then previously selected will never be unselected.

Reply:
ok I did a rename, and have reworked this so sessionmodel is always fresh against the most recent selections.